### PR TITLE
test: niporep: e2e testing

### PR DESCRIPTION
**Pull request by rvagg**
Replaces #12045

Initial form here has e2e onboarding of a single sector. Need to add:

* [ ] >1 sector in batch (note that real-proofs is quite expensive for this, so >1 sector might be slooow in CI)
* [x] Check `sector-activated` event(s) _(NOTE: needs fixing in builtin-actors: https://github.com/filecoin-project/builtin-actors/pull/1537/files/#r1648458930)_
* [x] Snap piece(s) into onboarded sector (and check events)
* [x] Error: 0 sectors
* [x] Error: >65 sectors (somehow .. maybe only with mock proofs)
* [x] Bad proof type, bad aggregate type (maybe)
* [ ] One bad proof, one good (maybe, somehow, see https://github.com/filecoin-project/builtin-actors/pull/1537#pullrequestreview-2123847997)
* [x] Bad sealer id, bad sealing number
* [x] Bad seal rand epoch (can't doo historical, we have a genesis chain and earliest is 180 days, but we can do future)
* [x] Bad proving deadline (current or next)

This branch uses https://github.com/filecoin-project/builtin-actors/pull/1537 for the actors bundle, but it also has https://github.com/filecoin-project/builtin-actors/pull/1553 applied on top of it because otherwise preseals don't onboard and we can't get the chain started. It also has https://github.com/filecoin-project/filecoin-ffi/pull/460 applied for filecoin-ffi because we need `verify-signature`.

### CODE REVIEW

**Comment by rvagg** [comment]
* Changed base to https://github.com/filecoin-project/lotus/pull/12143 until that's merged
* Rebased out the remaining pieces that have now been separately merged in to master
* Included https://github.com/filecoin-project/builtin-actors/releases/tag/v14.0.0-rc.1 bundle in here
* Squashed down to 2 commits - niporep test and actors bundle, nice and simple

**Comment by rvagg** [comment]
* Added `LOTUS_RUN_VERY_EXPENSIVE_TESTS` and put niporep w/ real-proofs in that category. It won't run in CI because it can't. We need to work on that. See https://github.com/filecoin-project/lotus/issues/12136
* I'm cutting scope on this particular PR to get it landed. What I want to add is multi-sector onboarding, currently in both mock and real proofs we only onboard a single sector and then snap a deal into it. I think I'd like to onboard 3 sectors and have one fail but the others pass. Or at least do 2. Or some combination that makes sense and doesn't make real-proofs take too long.

**Ready for review**

**Review by aarshkshah1992 — APPROVED** [review]
We can get the test for >1 sectors in a separate PR.

File: itests/niporep_manual_test.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+
+	head, err := miner.FullNode.ChainHead(ctx)
+	req.NoError(err)
+
+	actorIdNum, err := address.IDFromAddress(miner.ActorAddr)
+	req.NoError(err)
+	actorId := abi.ActorID(actorIdNum)
+
+	var provingDeadline uint64 = 7
+	if miner.IsUmmutableDeadline(ctx, provingDeadline) {
```
[the remark this answers is not in the document]
Comment (rvagg) [reply itests/niporep_manual_test.go:166]: ugh, I'm sure I fixed that already in the other one!

**Pull request merged**